### PR TITLE
[Flow] Add kafka Component 

### PR DESCRIPTION
**Pull request by oliver-zhang**
#### PR Description

Adds a flow component for kafka_exporter.

Closes https://github.com/grafana/agent/issues/2314

#### PR Checklist

- [x] CHANGELOG updated
- [x] Documentation added
- [x] Tests updated

### CODE REVIEW

File: docs/sources/flow/reference/components/prometheus.exporter.kafka.md
Comment (marctc) [inline docs/sources/flow/reference/components/prometheus.exporter.kafka.md:14]: this link gives a 404

File: component/prometheus/exporter/kafka/kafka.go
Comment (marctc) [inline component/prometheus/exporter/kafka/kafka.go:27]: These comments are not needed IMHO, but you can keep them if you want. Maybe at least start the comment with the name of the field?

**Review by marctc — COMMENTED** [review]
LGTM, thanks for the contribution. Added some minor comments. I'll defer to @gaantunes for another review

File: docs/sources/flow/reference/components/prometheus.exporter.kafka.md
Comment (oliver-zhang) [reply docs/sources/flow/reference/components/prometheus.exporter.kafka.md:14]: Yes, I will modify to the correct address

File: docs/sources/flow/reference/components/prometheus.exporter.kafka.md
Comment (spartan0x117) [inline docs/sources/flow/reference/components/prometheus.exporter.kafka.md:31]: nit: I think that the required arguments should be at the top of the list.
```suggestion
| `kafka_uris`                | `array(string)` | Address array (host:port) of Kafka server.                                                                                              |         | yes      |
| `instance`                  | `string`        |                                                                                                                                         |         | no       |
```

File: docs/sources/flow/reference/components/prometheus.exporter.kafka.md
Comment (spartan0x117) [inline docs/sources/flow/reference/components/prometheus.exporter.kafka.md:100]: Ideally examples are as copy-paste-able as we can make them:
```suggestion
// Configure a prometheus.scrape component to send metrics to
// prometheus.remote_write component.
prometheus.scrape "demo" {
  targets = [
    prometheus.exporter.kafka.example.targets
  ]
  forward_to = [prometheus.remote_write.demo.receiver]
}

prometheus.remote_write "demo" {
  endpoint {
    url = "http://mimir:9009/api/v1/push"

    basic_auth {
      username = "example-user"
      password = "example-password"
    }
  }
}
```

File: docs/sources/flow/reference/components/prometheus.exporter.kafka.md
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/prometheus.exporter.kafka.md:25]:
```suggestion
You can use the following arguments to configure the exporter's behavior.
```

Make it clear who can use the arguments and make sentence active.

File: docs/sources/flow/reference/components/prometheus.exporter.kafka.md
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/prometheus.exporter.kafka.md:31]:
```suggestion
| `instance`                  | `string`        | The`instance`label for metrics, default is the hostname:port of the first kafka_uris. If there is more than one string in kafka_uris, you must manually provide the instance value. |         | no       |
```

File: docs/sources/flow/reference/components/prometheus.exporter.kafka.md
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/prometheus.exporter.kafka.md:47]:
```suggestion
| `allow_concurrency`         | `bool`          | If set to true, all scrapes trigger Kafka operations. Otherwise, they will share results. WARNING: Disable this on large clusters.                                            | `true`  | no       |
```

File: docs/sources/flow/reference/components/prometheus.exporter.kafka.md
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/prometheus.exporter.kafka.md:48]:
```suggestion
| `max_offsets`               | `int`           | The maximum number of offsets to store in the interpolation table for a partition.                                                                                                      | `1000`  | no       |
```

File: docs/sources/flow/reference/components/prometheus.exporter.kafka.md
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/prometheus.exporter.kafka.md:58]: Why is Kafka in code here? And not just Kafka?

**Review by clayton-cornell — COMMENTED** [review]
Added some comments and suggestions

File: docs/sources/flow/reference/components/prometheus.exporter.kafka.md
Comment (oliver-zhang) [reply docs/sources/flow/reference/components/prometheus.exporter.kafka.md:58]: Yes, It's only collect kafka metrics, So the word `kafka` is unnecessary here

File: component/prometheus/exporter/kafka/kafka.go
Comment (marctc) [inline component/prometheus/exporter/kafka/kafka.go:27]: nit: remove blank lines?

**Review by marctc — APPROVED** [review]
LGTM, will merge when I have the thumbs up of @clayton-cornell

**Review by gaantunes — APPROVED** [review]
LGTM.

File: docs/sources/flow/reference/components/prometheus.exporter.kafka.md
Comment (clayton-cornell) [reply docs/sources/flow/reference/components/prometheus.exporter.kafka.md]: So should the text here be: 

> The targets that can be used to collect Kafka metrics.

(no code syntax, and capitalise Kafka)

File: docs/sources/flow/reference/components/prometheus.exporter.kafka.md
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/prometheus.exporter.kafka.md:44]:
```suggestion
| `zookeeper_uris`            | `array(string)` | Address array (hosts) of zookeeper server.                                                                                                                                        |         | no       |
```

File: docs/sources/flow/reference/components/prometheus.exporter.kafka.md
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/prometheus.exporter.kafka.md:43]:
```suggestion
| `use_zookeeper_lag`         | `bool`          | If set to true, use a group from zookeeper.                                                                                                                                         |         | no       |
```

File: docs/sources/flow/reference/components/prometheus.exporter.kafka.md
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/prometheus.exporter.kafka.md:41]:
```suggestion
| `insecure_skip_verify`      | `bool`          | If set to true, the server's certificate will not be checked for validity. This makes your HTTPS connections insecure.                                                                |         | no       |
```

**Review by clayton-cornell — APPROVED** [review]
Some additional small tweaks to the doc content. All good with  these chanegs.

**Comment by marctc** [comment]
@oliver-zhang could you fix the lint issues? I will merge it  after that, thanks!

**Comment by oliver-zhang** [comment]
> @oliver-zhang could you fix the lint issues? I will merge it after that, thanks!

It's resolved

**Pull request merged**
